src/main/java/org/javarosa/core/services/storage/IStorageUtilityIndexed.java

```diff
@@ -193,6 +193,23 @@
     */
     List<Integer> getIDsForValues(String[] metaFieldNames, Object[] values, LinkedHashSet<Integer> returnSet);
 
+    /**
+    * Retrieves a Vector of IDs of Externalizable objects in storage for which the field
+    * specified contains the values specified.
+    *
+    * @param metaFieldNames A list of metadata field names to match
+    * @param values     The values which must match the field names provided
+    * @param inverseFieldNames A list of field names to inverse match (!=)
+    * @param inverseValues     The values which the field must not match
+    * @param returnSet A LinkedHashSet of integers which match the return value
+    * @return A Vector of Integers such that retrieving the Externalizable object with any
```

Comment:
The behavior of this method is ambiguous if no keys are provided at all (IE: all incoming arrays are empty). Presumably, for instance, you can provide empty positive match fields, and a single negative match `a !='b'`, and get all the records where `a` doesn't equal `b`. From that logic, if you then remove the negative match you might expect to just get all of the records in the storage utility.

I'd clarify whether that's an error condition, or what to expect if not.

Reply:
Do you know what existing behavior is if no values are passed? I imagine that isn't something the code ever does (it doesn't make sense), but expect this to behave similarly.

Reply:
I believe the current code will return a full set iterator because it treats the query as a filter (Semantically: "Return all results which meet all filters" with an empty set of filters).